### PR TITLE
Metrics post was not pre-assigning a frame/message ID

### PR DESCRIPTION
**Pull request by itzg**

### CODE REVIEW

File: poller/connection_stream.go [hunk 1]
```diff
@@ -139,7 +139,9 @@ func (cs *EleConnectionStream) SendMetrics(crs *check.ResultSet) error {
 	}
 
 	if conn := cs.conns.ChooseBest(); conn != nil {
-		conn.GetSession().Send(check.NewMetricsPostRequest(crs, conn.GetClockOffset()))
+		request := check.NewMetricsPostRequest(crs, conn.GetClockOffset())
+		conn.GetSession().AssignFrameId(request)
```
Comment (itzg) [inline poller/connection_stream.go]: @rphillips  This usage is awkward and obviously error prone. I just now realized I should probably assign an ID in EleSession.Send if not set (i.e. zero).

**Comment by itzg** [comment]
Ready for a real review. Much smaller diff this time.

File: poller/connection_stream.go (hunk 1 above)
Comment (rphillips) [inline poller/connection_stream.go]: good idea

File: poller/session.go [hunk 2]
```diff
@@ -121,13 +121,15 @@ func NewSession(ctx context.Context, connection Connection, checksReconciler Che
...
 	request := protocol.NewHandshakeRequest(s.config)
-	request.SetId(&s.seq)
 	s.Send(request)
 }
 
 // Send stages a frame for sending after setting the target and source.
-// NOTE: The protocol.Frame.SetId MUST be called prior to this method.
+// NOTE: If the message's ID is not initialized an ID will be allocated.
 func (s *EleSession) Send(msg protocol.Frame) {
+	if msg.GetId() == 0 {
```
Comment (rphillips) [inline poller/session.go]: why only when it's zero?

**Comment by rphillips** [comment]
+1

File: poller/session.go (hunk 2 above)
Comment (itzg) [inline poller/session.go]: Discussed in IRC, but in short: to detect if the message ID had not be explicitly set prior to this point

**Pull request merged**